### PR TITLE
[devicelab] boot up benchmarks.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Enable Impeller benchmarks for drawAtlas/drawVertices on iOS/Metal, Android/GLES, and Android/Vulkan.

Enable impeller tessellation benchmarks on iOS/Metal and Android/Vulkan - not GLES as this is measuring backend agnostic performance.

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -2624,6 +2624,112 @@ targets:
...
+  - name: Linux_samsung_s10 path_tessellation_static_perf__timeline_summary
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    bringup: true
+    timeout: 60
+    properties:
+      ignore_flakiness: "true"
+      tags: >
+        ["devicelab", "android", "linux", "samsung", "s10"]
+      task_name: path_tessellation_static_perf__timeline_summary
```
Comment (gmackall) [inline .ci.yaml:2700]: I haven't added devicelab tests before, but maybe the issue is that the task names starting here (and in the next 3) don't line up with the file names like the previous ones do?

E.g. 
path_tessellation_static_perf__timeline_summary
vs
static_path_tessellation_perf__timeline_summary

Comment (jonahwilliams) [reply .ci.yaml:2700]: oh... that is probably it. 🤦 

Thank you!

File: TESTOWNERS [hunk 2]
```diff
@@ -94,6 +94,12 @@
 /dev/devicelab/bin/tasks/animated_blur_backdrop_filter_perf__timeline_summary.dart @jonahwilliams @flutter/engine
 /dev/devicelab/bin/tasks/animated_blur_backdrop_filter_perf_opengles__timeline_summary.dart @gaaclarke @flutter/engine
 /dev/devicelab/bin/tasks/slider_perf_android.dart @tahatesser @flutter/framework
+/dev/devicelab/bin/tasks/draw_vertices_perf_opengles__timeline_summary.dart @jonahwilliams @flutter/engine
+/dev/devicelab/bin/tasks/draw_atlas_perf_opengles__timeline_summary.dart @jonahwilliams @flutter/engine
+/dev/devicelab/bin/tasks/draw_vertices_perf__timeline_summary.dart @jonahwilliams @flutter/engine
+/dev/devicelab/bin/tasks/draw_atlas_perf__timeline_summary.dart @jonahwilliams @flutter/engine
+/dev/devicelab/bin/tasks/static_path_tessellation_perf__timeline_summary.dart @jonahwilliams @flutter/engine
```
Comment (keyonghan) [inline TESTOWNERS:101]: No ci.yaml target matching to this file?

File: TESTOWNERS [hunk 3]
```diff
@@ -94,6 +94,12 @@
...
+/dev/devicelab/bin/tasks/dynamic_path_tessellation_perf__timeline_summary.dart @jonahwilliams @flutter/engine
```
Comment (keyonghan) [inline TESTOWNERS:102]: Ditto.

File: .ci.yaml [hunk 4]
```diff
@@ -2624,6 +2624,112 @@ targets:
...
+  - name: Linux_samsung_s10 path_tessellation_dynamic_perf__timeline_summary
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    bringup: true
+    timeout: 60
+    properties:
+      ignore_flakiness: "true"
+      tags: >
+        ["devicelab", "android", "linux", "samsung", "s10"]
+      task_name: path_tessellation_dynamic_perf__timeline_summary
```
Comment (keyonghan) [inline .ci.yaml:2721]: Ditto.

File: .ci.yaml [hunk 5]
```diff
@@ -2624,6 +2624,112 @@ targets:
...
+
+  - name: Mac_ios path_tessellation_dynamic_ios__timeline_summary
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    timeout: 60
+    bringup: true
+    properties:
+      tags: >
+        ["devicelab", "ios", "mac"]
+      task_name: path_tessellation_dynamic_ios__timeline_summary
```
Comment (keyonghan) [inline .ci.yaml:2731]: Ditto.

File: .ci.yaml [hunk 6]
```diff
@@ -2624,6 +2624,112 @@ targets:
...
+    presubmit: false
+    bringup: true
+    timeout: 60
+    properties:
+      ignore_flakiness: "true"
+      tags: >
+        ["devicelab", "android", "linux", "samsung", "s10"]
+      task_name: draw_vertices_perf__timeline_summary
+
+  - name: Mac_ios draw_vertices_perf_ios__timeline_summary
```
Comment (keyonghan) [inline .ci.yaml:2649]: Nit: could you add targets to their platform set alphabetically.

File: .ci.yaml [hunk 7]
```diff
@@ -2624,6 +2624,112 @@ targets:
...
+    presubmit: false
+    bringup: true
+    timeout: 60
+    properties:
+      ignore_flakiness: "true"
+      tags: >
+        ["devicelab", "android", "linux", "samsung", "s10"]
+      task_name: draw_atlas_perf__timeline_summary
+
+  - name: Mac_ios draw_atlas_perf_ios__timeline_summary
```
Comment (keyonghan) [inline .ci.yaml:2681]: Nit: sort here as well.

File: .ci.yaml [hunk 8]
```diff
@@ -2624,6 +2624,112 @@ targets:
...
+    presubmit: false
+    bringup: true
+    timeout: 60
+    properties:
+      ignore_flakiness: "true"
+      tags: >
+        ["devicelab", "android", "linux", "samsung", "s10"]
+      task_name: path_tessellation_static_perf__timeline_summary
+
+  - name: Mac_ios path_tessellation_static_ios__timeline_summary
```
Comment (keyonghan) [inline .ci.yaml:2702]: Nit: sort here.

**Comment by jonahwilliams** [comment]
FYI @knopp

File: dev/devicelab/lib/tasks/perf_tests.dart [hunk 9]
```diff
@@ -672,6 +672,56 @@ TaskFunction createDrawPointsPerfTest({
...
+    '${flutterDirectory.path}/dev/benchmarks/macrobenchmarks',
+    'test_driver/run_app.dart',
+    'tessellation_perf_static',
+    enableImpeller: true,
+    testDriver: 'test_driver/path_tessellation_static_perf_test.dart',
+    saveTraceFile: true,
+  ).run;
+}
+
+TaskFunction createDynamicPathTessellationPerfTest() {
```
Comment (gaaclarke) [inline dev/devicelab/lib/tasks/perf_tests.dart:714]: nit: It's probably better if the name of these methods match the name of the test driver files, ie `createPathTessellationDynamicPerfTest`.

**Review by gaaclarke — APPROVED** [review]
lgtm modulo other reviews.  There is a lot of boilerplate to get this setup and I looked through to make sure i didn't see any typos.  I think it's good.  We might be getting to the point were we want to make this a bit easier to accomplish, where we specify something in 1 place and we get the 3 variants from it: opengles, vulkan, metal.

File: dev/devicelab/lib/tasks/perf_tests.dart (hunk 9 above)
Comment (jonahwilliams) [reply dev/devicelab/lib/tasks/perf_tests.dart:714]: Done

**Pull request merged**